test/db_test_helpers.py

```
@@ -377,6 +377,28 @@ def down_event_macs(cur) -> set:
+    """
+    cur = conn.cursor()
+    cur.execute("PRAGMA table_info(Devices)")
+    db_columns = {row[1] for row in cur.fetchall()}
+
+    cols = [k for k in device.keys() if k in db_columns]
+    placeholders = ", ".join("?" for _ in cols)
+    col_list = ", ".join(cols)
+    values = [device[c] for c in cols]
+
+    cur.execute(
+        f"INSERT OR IGNORE INTO Devices ({col_list}) VALUES ({placeholders})",
```

Ruff (0.15.17)

[error] Possible SQL injection vector through string-based query construction

(S608)